src/decorators/withBookmarks.jsx

```diff
@@ -34,12 +34,11 @@ const withBookmarks = Component => {
     const { anchor, emit, call } = props;
     // React state hooks
     const [bookmarks, setBookmarks] = useState({});
-    const firstKey = useMemo(() => Object.keys(bookmarks)[0] ?? false, [bookmarks]);
     const [active, setActive] = useState();
-    const renderedView = useMemo(
-      () => bookmarks[active || firstKey]?.view || [],
-      [active, bookmarks, firstKey]
-    );
+    const renderedView = useMemo(() => {
+      const firstKey = Object.keys(bookmarks)[0] ?? false;
+      return bookmarks[active || firstKey]?.view || [];
+    }, [active, bookmarks, firstKey]);
```

Comment:
How is this not throwing errors? You're setting `firstKey` in the dependency array whilst is now being created inside the useMemo... That means that `firstKey` doesn't exist outside of the function, how does this not trigger all sorts of errors? Did you test this?

Reply:
I did this in the meaning and did not get to test it. I thought David would grab it next.
Sorry.